modules/magniteAnalyticsAdapter.js

```diff
@@ -686,6 +698,52 @@ const handleBidWon = args => {
   addEventToQueue({ bidsWon: [bidWon] }, bidWon.renderAuctionId, 'bidWon');
 }
 
+const generateCountOfEvents = (missedEvents, pendingEvents) => {
```

Comment:
I don't really like the fact that this function is changing an input parameter. In terms of functional programming, which I don't know if it makes sense to bring the topic here, it would be desiderable to have it as a pure function. So, the best would be to create a new `missedEventsToReturn` that clones `missedEvents` and we'll return `missedEventsToReturn` instead. Why? Because we're actually relying on a side effect here, which is complicated to track down in case of bugs... 

This comments makes much more sense in a multithread environment, which is not javascript, so feel free to ignore it...

Reply:
yeah I understand, but I do not want to run a deep clone for each time this runs. 

These objects will be small so it is not a big deal though.

If you want me to I will update the code to clone and return the clone each time.

@musikele